src/api.rs

```diff
@@ -19,10 +20,20 @@ pub struct RegisterPlayerResponse {
 pub fn register_player(
     player_id_generator: State<PlayerIdGenerator>,
     scoreboard: State<Mutex<Scoreboard>>,
+    usernames: State<Mutex<Usernames>>,
     broadcaster: State<HostBroadcaster>,
 ) -> JSON<RegisterPlayerResponse>
 {
     let player_id = player_id_generator.next_id();
+    let username = PlayerNameGenerator.generate_username();
+
+    // Add the username to the Usernames map
+    let name_clone = {
+        let mut usernames = usernames.lock().expect("Usernames mutex was poisoned");
+        let old = usernames.insert(player_id, username.clone());
+        assert_eq!(None, old, "Player ID was registered twice");
+        username
+    };
```

Comment:
Ha, yeah. That was leftover from before I was ok with passing a `username.clone()` into the HashMap. For some reason my brain was telling me that was a bad idea